modin/core/execution/client/container.py

```diff
@@ -0,0 +1,742 @@
+# Licensed to Modin Development Team under one or more contributor license agreements.
+# See the NOTICE file distributed with this work for additional information regarding
+# copyright ownership.  The Modin Development Team licenses this file to you under the
+# Apache License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+"""Module contains ``ForwardingQueryCompilerContainer`` class."""
+
+import numpy as np
+import pandas
+from typing import Any, NamedTuple, Optional, Union
+from uuid import UUID, uuid4
+
+from modin.core.storage_formats.base.query_compiler import BaseQueryCompiler
+
+
+class ForwardingQueryCompilerContainer:
+    """
+    Container that forwards queries to query compilers within.
+
+    Parameters
+    ----------
+    query_compiler_class : type
+        Query compiler class to contain. Should be a subclass of BaseQueryCompiler.
+    io_class : type
+        The IO class to use for reading and writing data. Should be a subclass
+        of modin.core.io.io.BaseIO.
+    """
+
+    def __init__(self, query_compiler_class: type, io_class: type):
+        self._qc = {}
+        self._query_compiler_class = query_compiler_class
+        self._io_class = io_class
+
+    def _generate_id(self) -> UUID:
+        """
+        Generate an ID for a new query compiler.
+
+        Returns
+        -------
+        UUID
+            The generated ID.
+        """
+        id = uuid4()
+        while id in self._qc:
+            id = uuid4()
```

Comment:
Basically keep retrying until we ensure we have a unique ID? Might be helpful to have a comment for this 😄